### PR TITLE
Adding de2120-barcode-scanner

### PR DESCRIPTION
**Pull request by claydugo**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by claydugo** [comment]
@hmaarrfk, ready for review!

File: recipes/de2120-barcode-scanner/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://pypi.org/project/de2120-barcode-scanner/
+  summary: SparkFun Electronics DE2120 Barcode Scanner Breakout Package
+  license: MIT
+  license_file: LICENSE
```
Comment (hmaarrfk) [inline recipes/de2120-barcode-scanner/meta.yaml:36]: Can you include a link to the issue you opened upstream providing them with an example of how to include their LICENSE file in their pypi distribution:


https://github.com/psf/requests/blob/main/MANIFEST.in#L1
https://github.com/psf/requests/blob/main/setup.cfg#L2
They need to include
[metadata]
license_file = LICENSE





[17:45](https://ramonaoptics.slack.com/archives/D038AEV2QQ1/p1675118758602479)
from the setup.cfg
[17:46](https://ramonaoptics.slack.com/archives/D038AEV2QQ1/p1675118769882299)
and also have a [MANIFEST.in](http://manifest.in/) to add it to the "source distribution"
[17:46](https://ramonaoptics.slack.com/archives/D038AEV2QQ1/p1675118778727029)
the setup.cfg adds it to the "wheel" whl

Comment (claydugo) [reply recipes/de2120-barcode-scanner/meta.yaml:36]: https://github.com/sparkfun/DE2120_Py/issues/1

Comment (hmaarrfk) [reply recipes/de2120-barcode-scanner/meta.yaml:36]: Can you add a reference to this issue in the recipe so we know to delete our copy of their license file when it is resolved.

File: recipes/de2120-barcode-scanner/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/de2120_barcode_scanner-{{ version }}.tar.gz
+  sha256: 7d8e5fb88cd221213ea3b99493b0acdeb6d16655a1af393b466d4b4be66bf0e0
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
```
[the remark this answers is not in the document]
Comment (claydugo) [reply recipes/de2120-barcode-scanner/meta.yaml:17]: https://github.com/sparkfun/DE2120_Py/issues/2

Applied patch to address this and opened issue

**Comment by hmaarrfk** [comment]
Thanks @claydugo

**Pull request merged**